lib/rucio/core/replica.py

```diff
@@ -1785,12 +1790,19 @@ def update_replicas_states(replicas, nowait=False, session=None):
             values['tombstone'] = OBSOLETE
         elif replica['state'] == ReplicaState.AVAILABLE:
             rucio.core.lock.successful_transfer(scope=replica['scope'], name=replica['name'], rse_id=replica['rse_id'], nowait=nowait, session=session)
+            # If No locks we set a tombstone in the future
```

Comment:
why ? Is there a need to have an intermediate replica protection, here for 2 hours ?

Reply:
It's just added protection. The point is we need to add a tombstone since the replica would otherwise stay forever, since it has no rule. An immediate tombstone sounded too aggressive though, since the submission machinery could take too much time. In theory, the replica is protected by the sources table FK as well, but it's just added protection, since we know for sure that we do not have to delete this replica right away.

Reply:
Maybe getting this value from the internal config table instead would give some flexibility there. Default can be 2 hours if not defined.